espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/drivers/ComposeDriver.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.espressoserver.lib.drivers
+
+import androidx.compose.ui.semantics.SemanticsProperties
+import androidx.compose.ui.test.*
+import io.appium.espressoserver.EspressoServerRunnerTest
+import io.appium.espressoserver.lib.handlers.exceptions.InvalidElementStateException
+import io.appium.espressoserver.lib.handlers.exceptions.NoSuchElementException
+import io.appium.espressoserver.lib.handlers.exceptions.StaleElementException
+import io.appium.espressoserver.lib.helpers.*
+import io.appium.espressoserver.lib.model.*
+
+class ComposeDriver : AppDriver {
+    override val name = DriverContext.StrategyType.COMPOSE
+
+    override fun findElement(params: Locator): BaseElement {
+        val parentNodeInteraction = params.elementId?.let { getNodeInteractionById(it) }
+
+        val nodeInteractions = parentNodeInteraction?.findDescendantNodeInteractions(params)
+            ?: EspressoServerRunnerTest.composeTestRule.onAllNodes(semanticsMatcherForLocator(params))
+
+        if (nodeInteractions.fetchSemanticsNodes(false).isEmpty()) {
+            throw NoSuchElementException(
+                String.format(
+                    "Could not find compose element with strategy '%s' and selector '%s'",
+                    params.using, params.value
+                )
+            )
+        }
+        return ComposeElement(nodeInteractions[0])
+    }
+
+    override fun click(params: AppiumParams): Unit {
+        try {
+            getNodeInteractionById(params.elementId).performClick()
+        } catch (e: AssertionError) {
+            throw StaleElementException(params.elementId!!)
+        }
```

Comment:
I see that `performClick` internally uses [require](https://androidx.tech/artifacts/compose.ui/ui-test/1.0.2-source/androidx/compose/ui/test/InputDispatcher.kt.html), so we could also catch that exception here and rethrow InvalidElementState instead if click fails

Reply:
Added